kratos/controllers/temporal_controller.cpp

```diff
@@ -0,0 +1,158 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/variables.h"
+
+// Include base h
+#include "temporal_controller.h"
+
+namespace Kratos {
+
+Parameters TemporalController::GetDefaultParameters() const
+{
+    return Parameters(R"({
+        "model_part_name"     : "PLEASE_SPECIFY_THE_MODEL_PART_NAME",
+        "output_control_type" : "step",
+        "output_interval"     : 1.0
+    })" );
+}
+
+TemporalController::TemporalController(
+    const Model& rModel,
+    Parameters Settings)
+    : mpModel(&rModel)
+{
+    KRATOS_TRY
+
+    Settings.AddMissingParameters(GetDefaultParameters());
+
+    mModelPartName = Settings["model_part_name"].GetString();
+
+    const auto& output_control_type = Settings["output_control_type"].GetString();
+    if (output_control_type == "step") {

```

Comment:
Agree with the changes, but tbh, if we are going to derive different controllers from the base, I would move this to a different one (Time for time, StepController for steps, etc...).

But I would understand leaving this for historical reasons...

Reply:
It is for the historical reasons :/

Reply:
I thought the same. Maybe we can call it OutputController to make clear that this is for the output and save the Time keyword for an eventual time-based cotroller (same argument applies to the step keyword).

Reply:
Make sense. I will do it :)

Reply:
done :) can somebody approve this?